src/main/java/org/gridsuite/loadflow/server/service/LoadFlowService.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.loadflow.server.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.powsybl.commons.parameters.Parameter;
+import com.powsybl.commons.parameters.ParameterScope;
+import com.powsybl.loadflow.LoadFlowProvider;
+import org.apache.commons.lang3.tuple.Pair;
+import org.gridsuite.loadflow.server.dto.ComponentResult;
+import org.gridsuite.loadflow.server.dto.LoadFlowResult;
+import org.gridsuite.loadflow.server.dto.LoadFlowStatus;
+import org.gridsuite.loadflow.server.entities.ComponentResultEntity;
+import org.gridsuite.loadflow.server.entities.LoadFlowResultEntity;
+import org.gridsuite.loadflow.server.repositories.LoadFlowResultRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+@Service
+public class LoadFlowService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoadFlowService.class);
+
+    @Value("${loadflow.default-provider}")
+    private String defaultProvider;
+
+    private LoadFlowResultRepository resultRepository;
+
+    private ObjectMapper objectMapper;
+
+    NotificationService notificationService;
+
+    private UuidGeneratorService uuidGeneratorService;
+
+    public LoadFlowService(NotificationService notificationService, LoadFlowResultRepository resultRepository, ObjectMapper objectMapper, UuidGeneratorService uuidGeneratorService) {
+        this.notificationService = Objects.requireNonNull(notificationService);
```

Comment:
How is it possible that one of the bean is null ? Spring is supposed to check that itself no ?